src/handlers/records-write.ts

```diff
@@ -156,14 +160,51 @@ export class RecordsWriteHandler implements MethodHandler {
       status: { code: 202, detail: 'Accepted' }
     };
 
-    // delete all existing messages that are not newest, except for the initial write
+    // delete all existing messages of the same record that are not newest, except for the initial write
     await StorageController.deleteAllOlderMessagesButKeepInitialWrite(
       tenant, existingMessages, newestMessage, this.messageStore, this.dataStore, this.eventLog
     );
 
+    await this.postProcessingForCoreRecordsWrite(tenant, recordsWrite);
+
     return messageReply;
   };
 
+  private static validateSchemaForCoreRecordsWrite(recordsWriteMessage: RecordsWriteMessage, dataBytes: Uint8Array): void {
+    if (recordsWriteMessage.descriptor.protocol === PermissionsProtocol.uri) {
+      PermissionsProtocol.validateSchema(recordsWriteMessage, dataBytes);
+    }
+  }
+
+  /**
+   * Performs additional necessary tasks if the RecordsWrite handled is a core DWN RecordsWrite that need additional processing.
+   * For instance: a Permission revocation RecordsWrite.
+   */
+  private async postProcessingForCoreRecordsWrite(tenant: string, recordsWrite: RecordsWrite): Promise<void> {
+    // If this message is a Permission revocation, we need to delete all grant-authorized messages with timestamp after revocation
+    // TODO: https://github.com/TBD54566975/dwn-sdk-js/issues/716
+    // This code is a direct copy and paste from the original PermissionsRevokeHandler (no longer exists),
+    // but it appears that there was no test for it and it does not look like the code worked:
+    // - not seeing `permissionsGrantId` being an index
+    // - not seeing `this.dataStore` being called to delete actual data
+    // - test coverage is missing for the main delete logic
+    if (recordsWrite.message.descriptor.protocol === PermissionsProtocol.uri &&
+      recordsWrite.message.descriptor.protocolPath === PermissionsProtocol.revocationPath) {
+      const permissionsGrantId = recordsWrite.message.recordId;
```

Comment:
Sanity: I know this part isn't tested, but should this be the parentId of a revocation message?

Reply:
You are right. I'll change this now, but there are multiple other issues with this block that requires revisit by #716, e.g.
1. No deletion of actual data.
2. Atomic operation/failure resumption.